tok/hnsw/helper.go

```diff
@@ -444,6 +435,68 @@ func (ph *persistentHNSW[T]) createEntryAndStartNodes(
 	return entry, edges, nil
 }
 
+func encodeUint64MatrixUnsafe(matrix [][]uint64) []byte {
```

Comment:
If we do end up using it, we should draw a diagram of the format for better understanding.